### PR TITLE
Update and fix LGPL Z80 disassembler

### PR DESCRIPTION
**Pull request by Svenito**
Make the z80 disassembler more complete and fix incorrect outputs

Goes with radare/radare2-regressions#589

### CODE REVIEW

**Comment by Maijin** [comment]
Can you fix the build ?

File: libr/asm/arch/z80_cr/z80_cr.c [hunk 1]
```diff
@@ -142,15 +144,15 @@ static ut8 z80_ed_branch_index_res (ut8 hex) {
...
 		return hex-0x4e;
 	if (hex > 0x77 && 0x7c > hex)
-		return hex-0x51;
+		return hex-0x52;
 	if (hex > 0x9f && 0xa4 > hex)
-		return hex-0x75;
+		return hex-0x76;
 	if (hex > 0xa7 && 0xac > hex)
-		return hex-0x79;
+		return hex-0x7a;
```
Comment (radare) [inline libr/asm/arch/z80_cr/z80_cr.c]: spaces around - and use braces {} on every single if, even if its a oneliner

File: libr/asm/p/asm_z80.c [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #include <r_lib.h>
 #include <r_asm.h>
 
-#define GPL_COMPLIANT 0
+#define GPL_COMPLIANT 1
```
Comment (radare) [inline libr/asm/p/asm_z80.c]: maybe we should get rid of GPL_COMPLIANT completely, just remove the case in 0 and all that cpp work

File: libr/asm/arch/z80_cr/z80_cr.c [hunk 3]
```diff
@@ -127,30 +129,42 @@ static ut8 z80_fddd_branch_index_res (ut8 hex)
...
-		return hex-0x75;
-	if (hex > 0xa7 && 0xac > hex)
-		return hex-0x79;
-	if (hex > 0xaf && 0xb4 > hex)
-		return hex-0x7d;
-	if (hex > 0xb7 && 0xbc > hex)
-		return hex-0x81;
+	}
+	if (hex > 0x77 && 0x7c > hex) {
+		return hex-0x52;
```
Comment (radare) [inline libr/asm/arch/z80_cr/z80_cr.c]: add space around -

File: libr/asm/arch/z80_cr/z80_cr.c [hunk 4]
```diff
@@ -223,6 +243,7 @@ static int z80Disass (RAsmOp *op, const ut8 *buf, int len) {
 	if (!ret)
```
Comment (radare) [inline libr/asm/arch/z80_cr/z80_cr.c]: missing {}

File: libr/asm/arch/z80_cr/z80_cr.c [hunk 5]
```diff
@@ -248,18 +269,25 @@ static int z80Disass (RAsmOp *op, const ut8 *buf, int len) {
...
-			if (z_op[res].type == (Z80_OP16^Z80_ARG8))
+			}
+			if (z_op[res].type == (Z80_OP16^Z80_ARG8)) {
 				sprintf (op->buf_asm, z_op[res].name, buf[2], buf[3]);
+			}
 			if (z_op[res].type == (Z80_OP24^Z80_ARG8)) {
 				cb_tab = (char **) z_op[res].op_moar;
 				sprintf (op->buf_asm, cb_tab[z80_op_24_branch_index_res (buf[3])], buf[2]);
 			}
+			if (z_op[res].type == (Z80_OP16^Z80_ARG8^Z80_ARG16)) {
```
Comment (radare) [inline libr/asm/arch/z80_cr/z80_cr.c]: spaces around ^

File: libr/asm/p/asm_z80.c (hunk 2 above)
Comment (radare) [inline libr/asm/p/asm_z80.c]: ok!

**Pull request merged**